Sources/CoreLocationMiddleware/CoreLocationMiddleware.swift

```diff
@@ -67,43 +79,76 @@ public final class CoreLocationMiddleware: Middleware {
         delegate.output = output
         manager.delegate = delegate
     }
+    
     public func handle(action: LocationAction, from dispatcher: ActionSource, afterReducer: inout AfterReducer) {
         switch action {
-        case .startMonitoring, .authorized:
-            startMonitoring()
-        case .stopMonitoring:
-            stopMonitoring()
+        case .startMonitoring: startMonitoring()
+        case .stopMonitoring: stopMonitoring()
+        case .requestAuthorizationStatus:
+            if #available(iOS 14.0, *) {
+                delegate.output?.dispatch(getAuthzStatus(status: manager.authorizationStatus))
+            } else {
+                return
+            }
+        case .requestAuthorizationType:
+            switch getState?().authzType {
+            case .always:
+                manager.requestAlwaysAuthorization()
+            case .whenInUse:
+                manager.requestWhenInUseAuthorization()
+            case .none:
+                break
+            }
+        case .requestPosition: manager.requestLocation()
+        case .requestDeviceCapabilities: delegate.output?.dispatch(getDeviceCapabilities())
         default: return
         }
     }
+    
     func startMonitoring() {
-        switch getState?() {
-        case .authorized?:
-            manager.startUpdatingLocation()
-        default:
-            // requestAlwaysAuthorization or requestWhenInUseAuthorization could be decided on
-            // the middleware init or as payload for startMonitoring
-            manager.requestAlwaysAuthorization()
+        let stateType = getState?().authzType
+        let stateStatus = getState?().authzStatus
+        
+        switch stateType {
+        case .always:
+            if CLAuthorizationStatus.authorizedAlways != stateStatus {
+                manager.requestAlwaysAuthorization()
+            }
+        case .whenInUse:
+            if ![CLAuthorizationStatus.authorizedAlways, CLAuthorizationStatus.authorizedWhenInUse].contains(stateStatus) {
+                manager.requestWhenInUseAuthorization()
```

Comment:
Should we return in this case, to avoid manager.startUpdatingLocation() when we don't have any permissions?

Reply:
See #6 to further discuss.